tests/unit/test_charm.py

```diff
@@ -862,7 +865,11 @@ def test_get_secret_secrets(harness, scope):
 
 
 @patch_network_get(private_address="1.1.1.1")
-def test_set_secret(harness):
+def test_set_secret_in_databag(harness, juju_has_secrets):
+    # As this test set secrets using set_secret and checks if they end up
+    # inside the relation databag, this is meant to run on juju2 only
+    if juju_has_secrets:
+        return
```

Comment:
We shouldn't have test returning on a condition.

`pytest.makr.skip` was invented for this reason :-)

Could you pls use something like my fixtures instead? 
https://github.com/canonical/data-platform-libs/blob/main/tests/conftest.py#L37-L51